mappings/bloblang/AWS/CloudTrail/cloudtrail_authentication.blobl

```diff
@@ -4,61 +4,96 @@ meta ocsf_topic = "authentication"
 root.class_uid = 3002
 root.class_name = "Authentication"
 root.category_uid = 3
-root.category = "Identity & Access Management Category"
+root.category_name = "Identity & Access Management"
 root.cloud.provider = "AWS"
 root.severity_id = 1
 root.severity = "Informational"
-root.status_id = match this.ConsoleLogin.responseElements.ConsoleLogin {
+root.status_id = match this.responseElements.ConsoleLogin|serviceEventDetails.UserAuthentication {
         "Unknown" => 0
 	    "Success" => 1
 	    "Failure" => 2
 	    "Other" => 99
-    }
-root.status = this.ConsoleLogin.responseElements.ConsoleLogin
+	    _ => null
+   }
+root.status = this.responseElements.ConsoleLogin|serviceEventDetails.UserAuthentication
 root.activity_id = match {
         this.eventName == "ConsoleLogin" => 1
         this.eventName == "ExternalIdPDirectoryLogin" => 1
+        this.eventName == "UserAuthentication" => 1
         this.eventName.contains("AssumeRole") => 1
         _ => 99
     }
 root.activity_name = match {
         this.eventName == "ConsoleLogin" => "Logon"
         this.eventName == "ExternalIdPDirectoryLogin" => "Logon"
+        this.eventName == "UserAuthentication" => "Logon"
         this.eventName.contains("AssumeRole") => "Logon"
         _ => "Other"
     }
-root.time = this.eventTime
-root.metadata.version = this.eventVersion
+root.type_uid = match {
```

Comment:
type_uid = root.class_uid.number() * 100 + root.activity_id.number()
type_name = root.class_name + ": " + root.activity_name

This may be a better transformation